Document the process for disabling workflows (#31603)
cilium-team/cilium: Documentation/contributing/testing/ci.rst

* ``Flake, DNS not ready, #3333``
* ``CI-Bug, K8sValidatedPolicyTest: Namespaces, pod not ready, #9939``
* ``Regression, k8s host policy, #1111``

Disabling Github Actions Workflows
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. warning::
    Do not use the `GitHub web UI <https://docs.github.com/en/actions/using-workflows/disabling-and-enabling-a-workflow?tool=webui>`_
    to disable GitHub Actions workflows. It makes it difficult to find out who
    disabled the workflows and why.

Alternatives to Disabling Github Actions Workflows
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Before proceeding, consider the following alternatives to disabling an entire
GitHub Actions workflow.

- Skip individual tests. If specific tests are causing the workflow to fail,
  disable those tests instead of disabling the workflow. When you disable a
  workflow, all the tests in the workflow stop running. This makes it easier
  to introduce new regressions that would have been caught by these tests
  otherwise.

> - Remove the workflow from the list of required status checks. This way the

Reviewer: How?

(In fact the answer is in Step 2: Contact the maintainers. But it would be useful to mention it or link to Step 2 here.)

Author: updated ✨ https://deploy-preview-31603--docs-cilium-io.netlify.app/contributing/testing/ci#disabling-github-actions-workflows

  workflow still runs on pull requests, but you can still merge them without
  the workflow succeeding.

Step 1: Open a GitHub Issue
^^^^^^^^^^^^^^^^^^^^^^^^^^^

Open a GitHub issue to track activities related to fixing the workflow. If there
are existing test flake GitHub issues, list them in the tracking issue. Find an
assignee for the tracking issue to avoid the situation where the workflow remains
disabled indefinitely because nobody is assigned to actually fix the workflow.

Step 2: Update the required status check list
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If the workflow is in the required status check list, it needs to be remove it
from the list. Notify the `cilium-maintainers team <https://github.com/orgs/cilium/teams/cilium-maintainers>`_
by mentioning ``@cilium/cilium-maintainers`` in the tracking issue and ask them
to remove the workflow from the required status check list.

Step 3: Update the workflow configuration
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Update the workflow configuration by opening a pull request, getting it reviewed,
> and merging it.

Reviewer: The formatting of the headings for `Step 3` and the sub-steps is nearly identical, so that when looking at the preview, I initially thought they were all separate steps and I had no idea what update Step 3 was referring to. I'd introduce the sub-steps to make it clearer.

Another nit: Not everyone can merge PRs themselves.

```suggestion
Update the workflow configuration as described in the following sub-steps. Open a pull request with your changes, have it reviewed, then merged.
```

Author: should we use tabs here? https://deploy-preview-31603--docs-cilium-io.netlify.app/contributing/testing/ci#disabling-github-actions-workflows looks a bit more readable 💭

Disabling Github Actions Workflows
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. warning::
    Do not use the `GitHub web UI <https://docs.github.com/en/actions/using-workflows/disabling-and-enabling-a-workflow?tool=webui>`_
    to disable GitHub Actions workflows. It makes it difficult to find out who
    disabled the workflows and why.

Alternatives to Disabling Github Actions Workflows
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Before proceeding, consider the following alternatives to disabling an entire
GitHub Actions workflow.

- Skip individual tests. If specific tests are causing the workflow to fail,
  disable those tests instead of disabling the workflow. When you disable a
  workflow, all the tests in the workflow stop running. This makes it easier
  to introduce new regressions that would have been caught by these tests
  otherwise.
- Remove the workflow from the list of required status checks. This way the
  workflow still runs on pull requests, but you can still merge them without
  the workflow succeeding. To remove the workflow from the required status check
  list, post a message in the `#testing Slack channel <https://cilium.slack.com/archives/C7PE7V806>`_

>   and @mention people in the the `cilium-maintainers team <https://github.com/orgs/cilium/teams/cilium-maintainers>`_.

Reviewer: ```suggestion
  and @mention people in the `cilium-maintainers team <https://github.com/orgs/cilium/teams/cilium-maintainers>`_.
```

Step 1: Open a GitHub Issue
^^^^^^^^^^^^^^^^^^^^^^^^^^^

Open a GitHub issue to track activities related to fixing the workflow. If there
are existing test flake GitHub issues, list them in the tracking issue. Find an
assignee for the tracking issue to avoid the situation where the workflow remains
disabled indefinitely because nobody is assigned to actually fix the workflow.

Step 2: Update the required status check list
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If the workflow is in the required status check list, it needs to be remove it

Reviewer: ```suggestion
If the workflow is in the required status check list, it needs to be removed
```
